### PR TITLE
ci: bump node from v16 to v20 LTS

### PR DESCRIPTION
**Pull request by ndv99**
## Done
- Changed node version to 20.X in CI pipeline
- Added `.nvmrc` with latest node LTS for nvm users

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v20.9.0
```
Comment (petermakowski) [inline .nvmrc:1]: How about we specify the major version only to match test.yml workflow?

```suggestion
v20
```

Comment (ndv99) [reply .nvmrc:1]: Let me test that locally to see if it works as intended. One sec

Comment (ndv99) [reply .nvmrc:1]: Yep, that works!

Comment (petermakowski) [reply .nvmrc]: Even better if we centralised this and read the node version from .nvmrc file in the github action: https://github.com/actions/setup-node/issues/32#issuecomment-1003854758

Comment (petermakowski) [reply .nvmrc]: @ndv99  This is more relevant for us: https://github.com/actions/setup-node/issues/32#issuecomment-872375574

**Comment by ndv99** [comment]
oh hey that worked

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -6,12 +6,19 @@ on:
...
+      - name: "Read .nvmrc"
+        run: "echo NVMRC=`cat .nvmrc` >> $GITHUB_ENV"
   lint:
     name: Lint
+    needs: read-node-version
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        node-version: [16.x]
+        node-version: [20.x]
```
Comment (petermakowski) [inline .github/workflows/test.yml]: This can be removed since we use a single version of node specified in the nvmrc.

Comment (ndv99) [reply .github/workflows/test.yml]: Pushed a commit resolving this :+1:

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -6,12 +6,16 @@ on:
   pull_request:
     types: [opened, synchronize, reopened]
 jobs:
+  read-node-version:
+    name: "Read NodeJS version"
+    runs-on: ubuntu-22.04
+    steps:
+      - name: "Read .nvmrc"
+        run: "echo NVMRC=`cat .nvmrc` >> $GITHUB_ENV"
```
Comment (petermakowski) [inline .github/workflows/test.yml:14]: You need to checkout the repo before reading the version in order for the file to be available for the runner.

Comment (petermakowski) [reply .github/workflows/test.yml:14]: https://github.com/canonical/maas-react-components/blob/main/.github/workflows/CI.yml

**Review by petermakowski — APPROVED** [review]
Much cleaner, nice one!

**Pull request merged**